superset/security/manager.py

```diff
@@ -1520,3 +1517,70 @@ def has_guest_access(self, dashboard: "Dashboard") -> bool:
             if str(resource["id"]) == str(dashboard.embedded[0].uuid):
                 return True
         return False
+
+    def raise_for_ownership(self, resource: Model) -> None:
+        """
+        Raise an exception if the user does not own the resource.
+
+        Note admins are deemed owners of all resources.
+
+        :param resource: The dashboard, dataste, chart, etc. resource
+        :raises SupersetSecurityException: If the current user is not an owner
+        """
+
+        # pylint: disable=import-outside-toplevel
+        from superset import db
+
+        if self.is_admin():
+            return
+
+        orig_resource = (
+            db.session.query(resource.__class__).filter_by(id=resource.id).first()
```

Comment:
nit: `db.session.query(resource.__class__).filter_by(id=resource.id).one_or_none()` or `db.session.query(resource.__class__).get(resource.id)`

Reply:
@dpgaspar I agree with the `one_or_none`. This was a copy-and-paste from `check_ownership` but I'll make the update.